### DISCUSSION TITLE
How to build multiple entries with different targets

### DISCUSSION THREAD

**Discussion by longht021189** [web-infra-dev/rsbuild#1565] [Q&A] [answered]
I'd like to build multiple entries with:
- Different targets: `node` and `web`.
- Switch alias configs or different entry files.

But Rsbuild only provides multiple targets in output.

**Comment by chenjiahan** [comment, marked as answer]
You can use tools.rspack to set different entries or aliases for different targets. We will support this feature in the Rsbuild configuration later.

```js
export default defineConfig({
  output: {
    targets: ['web', 'node']
  },
  tools: {
    rspack(config, { target }) {
      if (target === 'web') {
        config.entry = {};
        config.resolve!.alias = {};
      } else {
        // ...
      }
    }
  }
});
```

**Reply by longht021189** [reply]
Thank you!
